packages/devtools_app/lib/src/screens/inspector/easy_image_provider.dart

```diff
@@ -0,0 +1,52 @@
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+
+class EasyImageProvider extends ImageProvider<EasyImageProvider> {
+  const EasyImageProvider(this.data, this.key, { this.scale = 1.0 });
+
+  /// The file to decode into an image.
+  final Uint8List data;
+
+  /// The scale to place in the [ImageInfo] object of the image.
+  final double scale;
+
+  final String? key;
+
+  @override
+  ImageStreamCompleter load(EasyImageProvider key, decode) {
+    return MultiFrameImageStreamCompleter(
+      codec: _loadAsync(key, decode),
+      scale: key.scale,
+      debugLabel: this.key,
+    );
+  }
+
+  Future<ui.Codec> _loadAsync(EasyImageProvider key, DecoderCallback decode) async {
+    assert(key == this);
+
+    return decode(data);
+  }
+
+  @override
+  Future<EasyImageProvider> obtainKey(ImageConfiguration configuration) {
+    return SynchronousFuture<EasyImageProvider>(this);
+  }
+
+  @override
```

Comment:
nit: why are we overriding == and hashcode?

Reply:
Thanks. The implementation of this component is in another function big picture detection, and may refer to the implementation of other providers. Here is directly reused. It may not be needed here. i will delete it.